framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -721,7 +721,10 @@ protected void checkConstructorResult(
                     qualifierHierarchy.findAnnotationInHierarchy(constructorAnnotations, top);
             if (!qualifierHierarchy.isSubtype(top, constructorAnno)) {
                 checker.report(
-                        Result.warning("inconsistent.constructor.type", constructorAnno),
+                        Result.warning(
+                                "inconsistent.constructor.type",
+                                constructorAnno,
+                                qualifierHierarchy.getTopAnnotation(constructorAnno)),
```

Comment:
Shouldn't this just be `top`? Or am I missing something?

Reply:
@abslamp Can you fix this?

Reply:
Thanks for feedback. It's now fixed.